stl/inc/format

```diff
@@ -2934,15 +3030,54 @@ struct formatter<basic_string_view<_CharT, _Traits>, _CharT>
 
 template <class _CharT, class... _Args>
 struct _Basic_format_string {
+    static constexpr size_t _Num_args = sizeof...(_Args);
+
     basic_string_view<_CharT> _Str;
+    size_t _Arg_use_count[_Num_args > 0 ? _Num_args : 1]{};
+    size_t _Estimated_size    = 0;
+    bool _Is_estimation_exact = false;
 
     template <class _Ty>
         requires convertible_to<const _Ty&, basic_string_view<_CharT>>
     consteval _Basic_format_string(const _Ty& _Str_val) : _Str(_Str_val) {
         if (_Is_execution_charset_utf8()) {
+#ifndef __clang__ // TRANSITION, clang consteval bug (likely https://github.com/llvm/llvm-project/issues/52648)
+            _Format_checker<_CharT, remove_cvref_t<_Args>...> _Handler{_Str};
+            _Parse_format_string(_Str, _Handler);
+            _RANGES copy(_Handler._Arg_use_count, _Arg_use_count);
+            _Estimated_size      = _Handler._Estimated_size;
+            _Is_estimation_exact = _Handler._Is_estimation_exact;
+#else // ^^^ no workaround ^^^ / vvv workaround vvv
             _Parse_format_string(_Str, _Format_checker<_CharT, remove_cvref_t<_Args>...>{_Str});
+            _Estimated_size = _Str.size() + _Num_args * 8;
+            _STD fill(_STD begin(_Arg_use_count), _STD end(_Arg_use_count), static_cast<size_t>(1));
+#endif // ^^^ workaround ^^^
+        } else {
+            // fallback to assumption that all arguments are used once
+            _Estimated_size = _Str.size() + _Num_args * 8;
+            _STD fill(_STD begin(_Arg_use_count), _STD end(_Arg_use_count), static_cast<size_t>(1));

```

Comment:
This uses `_STD fill()` with `_STD begin()` and `_STD end()`. Would using `_RANGES fill()` be simpler, and more consistent with the use of `_RANGES copy()` immediately above?